### PR TITLE
Fix small issues

### PR DESCRIPTION
**Pull request by nightwing**
fixes https://github.com/ajaxorg/ace/issues/3195 https://github.com/ajaxorg/ace/issues/3196 https://github.com/ajaxorg/ace/issues/731 https://github.com/ajaxorg/ace/issues/36 https://github.com/ajaxorg/ace/issues/2803 https://github.com/ajaxorg/ace/issues/1667
@lennartcl

### CODE REVIEW

File: lib/ace/commands/default_commands.js [hunk 1]
```diff
@@ -622,7 +622,7 @@ exports.commands = [{
     scrollIntoView: "cursor"
 }, {
     name: "transposeletters",
-    bindKey: bindKey("Ctrl-T", "Ctrl-T"),
+    bindKey: bindKey("Alt-X", "Ctrl-T"),
```
Comment (lennartcl) [inline lib/ace/commands/default_commands.js]: Should that even have a keybinding?

File: lib/ace/mode/json_highlight_rules.js [hunk 2]
```diff
@@ -57,12 +57,16 @@ var JsonHighlightRules = function() {
                 token : "constant.language.boolean",
                 regex : "(?:true|false)\\b"
             }, {
-                token : "invalid.illegal", // single quoted strings are not allowed
+                token : "text", // single quoted strings are not allowed
                 regex : "['](?:(?:\\\\.)|(?:[^'\\\\]))*?[']"
             }, {
-                token : "invalid.illegal", // comments are not allowed
+                token : "comment", // comments are not allowed, but who cares?
```
Comment (lennartcl) [inline lib/ace/mode/json_highlight_rules.js]: 😆  (seems like a good idea)

File: lib/ace/worker/worker_client.js [hunk 3]
```diff
@@ -36,7 +36,29 @@ var net = require("../lib/net");
...
+        blobBuilder.append(script);
+        return blobBuilder.getBlob("application/javascript");
+    }
+}
+
+function createWorker(workerUrl) {
+    var blob = $workerBlob(workerUrl);
+    var URL = window.URL || window.webkitURL;
+    var blobURL = URL.createObjectURL(blob);
+    // calling URL.revokeObjectURL before worker is terminated breaks it on IE Edge
```
Comment (lennartcl) [inline lib/ace/worker/worker_client.js]: Also won't matter much for our use case.

File: lib/ace/commands/default_commands.js (hunk 1 above)
Comment (nightwing) [inline lib/ace/commands/default_commands.js]: for multiselect it transposes selected word, which is more useful that teansposing letters

**Pull request merged**